# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1510164178 18000
#      Wed Nov 08 13:02:58 2017 -0500
# Node ID d26b8f30ee4bb789be07123c7422266b1ba46b58
# Parent  92acdcf72c462bdbfb3a3f64c9195055391e05ea
fix lengths in .ofig suffix comparison

* Figure.cc (Figure::save_figure_callback): Fix substring extraction.
* __add_default_menu__.m (save_cb): Likewise.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -896,17 +896,17 @@ namespace QtHandles
   void
   Figure::save_figure_callback (const std::string& file)
   {
     figure::properties& fp = properties<figure> ();
     octave_value fnum = fp.get___myhandle__ ().as_octave_value ();
 
     size_t flen = file.length ();
 
-    if (flen > 4 && file.substr (flen-4, 4) == ".ofig")
+    if (flen > 5 && file.substr (flen-5) == ".ofig")
       Ffeval (ovl ("hgsave", fnum, file));
     else
       Ffeval (ovl ("print", fnum, file));
   }
 
   void
   Figure::copy_figure_callback (const std::string& format)
   {
diff --git a/scripts/plot/util/private/__add_default_menu__.m b/scripts/plot/util/private/__add_default_menu__.m
--- a/scripts/plot/util/private/__add_default_menu__.m
+++ b/scripts/plot/util/private/__add_default_menu__.m
@@ -83,17 +83,17 @@ function __save_as__ (caller)
       "*.eps;*.epsc;*.pdf;*.svg;*.ps;*.tikz", "Vector Image Formats";
       "*.gif;*.jpg;*.png;*.tiff", "Bitmap Image Formats"},
      "Save Figure", fullfile (pwd, "untitled.ofig"));
 
   if (filename != 0)
     fname = fullfile (filedir, filename);
     set (gcbf, "filename", fname);
     flen = numel (fname);
-    if (flen > 4 && strcmp (fname(flen-3:end), ".ofig"))
+    if (flen > 5 && strcmp (fname(flen-4:end), ".ofig"))
       hgsave (caller, fname);
     else
       saveas (caller, fname);
     endif
   endif
 endfunction
 
 
